elements/pfe-switch/src/pfe-switch.js

```diff
@@ -0,0 +1,178 @@
+import PFElement from "../../pfelement/dist/pfelement.js";
+
+class PfeSwitch extends PFElement {
+  static get tag() {
+    return "pfe-switch";
+  }
+
+  get schemaUrl() {
+    return "pfe-switch.json";
+  }
+
+  get templateUrl() {
+    return "pfe-switch.html";
+  }
+
+  get styleUrl() {
+    return "pfe-switch.scss";
+  }
+
+  static get events() {
+    return {
+      change: `${this.tag}:change`
+    };
+  }
+
+  get pfeChecked() {
+    return this.getAttribute("pfe-checked");
+  }
+
+  // Declare the type of this component
+  static get PfeType() {
+    return PFElement.PfeTypes.Content;
+  }
+
+  constructor() {
+    super(PfeSwitch, { type: PfeSwitch.PfeType });
+    this._updateSwitchState = this._updateSwitchState.bind(this);
+    this.settings = {};
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    const $pfeSwitch = this.shadowRoot.querySelector(".pfe-switch");
+
+    // Get settings from attributes
+    const settingAttributes = [
+      "pfe-message-on",
+      "pfe-message-off",
+      "pfe-hide-label"
+    ];
+    for (let index = 0; index < settingAttributes.length; index++) {
+      const settingAttribute = settingAttributes[index];
+      // Adds settings to an object and lops of `data-` from the index
+      this.settings[settingAttribute.substr(4)] = this.getAttribute(
+        settingAttribute
+      );
+    }
+
+    // Get a reference to the checkbox and label that should be slotted
+    this.checkbox = null;
```

Comment:
I love this set-up.  I would just update the name of your variables to remove the jQuery syntax and use `this._foo` convention instead and it's +1.